### PR TITLE
Split bitcoinz daemon into separate docker container

### PR DESCRIPTION
**Pull request by renuzit**
This commit splits the bitcoinz-insight-patched daemon into a separate
docker container. This allows for more modularity and better support for
automated builds.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,15 +1,25 @@
 version: "3"
 
 services:
+  bitcoinz-insight-patched:
+    container_name: bitcoinz-insight-patched
```
Comment (egyptianbman) [inline docker-compose.yml]: I'm not a fan of containers with a name. Since you're `link`ing this below, this isn't necessary.

Comment (renuzit) [reply docker-compose.yml]: Link is required so that the server name is resolvable by DNS. Do you have a better suggestion?

Comment (egyptianbman) [reply docker-compose.yml]: Link is required, but `container_name` shouldn't be.

File: bitcoinz.conf [hunk 2]
```diff
@@ -14,13 +14,13 @@ addnode=dnsseed.kemperink.org
 addnode=btcz.es
 addnode=btcz.vnminers.com
 server=1
-whitelist=127.0.0.1
+rpcallowip=0.0.0.0/0
```
Comment (egyptianbman) [inline bitcoinz.conf]: Does binding this to `bitcoinz-insight-patched/32` work instead? If someone runs this without a firewall, I can see it leading to potential security concerns.

File: docker-compose.yml [hunk 3]
```diff
@@ -1,14 +1,24 @@
...
+      - bitcoinz-insight-patched
     network_mode: bridge
     restart: always
     ports:
       - 3001:3001
-      - 1989:1989
     volumes:
-      - ./data/:/bitcoinz/data
-      - ./bitcoinz.conf:/bitcoinz/data/zcash.conf
+      - ./pools.json:/usr/src/app/btcz-explorer/node_modules/insight-api-btcz/pools.json
```
Comment (egyptianbman) [inline docker-compose.yml]: This shouldn't be needed any longer.

File: bitcoinz.conf (hunk 2 above)
Comment (renuzit) [reply bitcoinz.conf]: Changed it to `rpcallowip=172.16.0.0/12`

File: docker-compose.yml (hunk 3 above)
Comment (renuzit) [reply docker-compose.yml]: Removed, thanks!

File: README.md [hunk 4]
```diff
@@ -19,28 +20,16 @@ docker build -t btcz/bitcoinz-rest-api .
...
-```
-sudo mkdir -p /opt/storage/bitcoinz/data/
-sudo chmod a+rwx /opt/storage/bitcoinz/data/
-cp bitcoinz.conf /opt/storage/bitcoinz/data/zcash.conf
 docker run -it --name bitcoinz-rest-api \
+  --link bitcoinz-insight-patched
   -p 3001:3001 \
-  -p 1989:1989 \
-  -v /opt/storage/bitcoinz/data/:/bitcoinz/data \
   btcz/bitcoinz-rest-api
```
Comment (egyptianbman) [inline README.md:33]: Is this right? I'm getting:
```
Pulling insight (btcz/bitcoinz-rest-api:latest)...                             
ERROR: pull access denied for btcz/bitcoinz-rest-api, repository does not exist or may require 'docker login'                                                 
```

Comment (renuzit) [reply README.md:33]: No, that's not right. You need to build it first I guess.

Comment (egyptianbman) [reply README.md:33]: I'm using docker-compose so this isn't an issue for me but I think it'll cause confusion. Do you want me to push up a fix or do you want to?

Comment (egyptianbman) [reply README.md:33]: Let me know if you need me to push up the update to the README.

Comment (renuzit) [reply README.md:33]: Do it!

**Review by egyptianbman — APPROVED** [review]
Actually, as I'm updating it, I do see the build step in there already. Ignore me, I need more sleep. Please merge at will; thanks!

**Pull request merged**